site/app/controllers/AuthenticationController.php

```diff
@@ -72,6 +72,10 @@ public function logout() {
      * @return MultiResponse
      */
     public function loginForm($old = null) {
+        if (!is_null($old) && !Utils::startsWith(urldecode($old), $this->core->getConfig()->getBaseUrl())) {
```

Comment:
Setting the header is enough to fix this problem, moving it to `index.php` like @MasterOdin suggested should fix the issue of using iframes to use cross-origin hacks on Submitty.

I don't think there's any point of this check once that's in place, the `SAME-ORIGIN` policy will perform the same check and most likely will be more effective than this.

Reply:
`SAME-ORIGIN` policy will not stop redirects to other domains.

Reply:
This confusion is my bad, I should have split this up into 2 prs. The header line is to stop iframes from working with submitty. All the other lines are to stop the login redirect from sending you to another website after you log in. They are semi unrelated issues and require different fixes.

Reply:
It should be only if they run their personal site at the same origin as Submitty, which is that `https://submitty.cs.rpi.edu` has a different origin than `https://rpi.edu/~pevelm`. But given that no one is using iframes, fine to just deny for now and revisit if we have the need.